specification/hardwaresecuritymodules/HardwareSecurityModules.Management/CloudHsmCluster.tsp

```diff
@@ -0,0 +1,240 @@
+
+  /**
+   * The List operation gets information about the Cloud HSM Clusters associated with the subscription and within the specified resource group.
+   */
+  listByResourceGroup is ArmResourceListByParent<
+    CloudHsmCluster,
+    Parameters = {
+      /**
+       * The page-continuation token to use with a paged version of this API
+       */
+      @query("$skiptoken")
+      $skiptoken?: string;
+    }
+  >;
+
+  /**
+   * The List operation gets information about the Cloud HSM Clusters associated with the subscription.
+   */
+  listBySubscription is ArmListBySubscription<
+    CloudHsmCluster,
+    Parameters = {
+      /**
+       * The page-continuation token to use with a paged version of this API
+       */
+      @query("$skiptoken")
+      $skiptoken?: string;
+    }
+  >;
+
+  /**
+   * Pre Backup operation to validate whether the customer can perform a backup on the Cloud HSM Cluster resource in the specified subscription.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  validateBackupProperties is ArmResourceActionAsync<
+    CloudHsmCluster,
+    BackupRequestProperties,
+    ArmResponse<BackupResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = BackupResult> &
+      Azure.Core.Foundations.RetryAfterHeader,
+    OptionalRequestBody = true
+  >;
+
+  /**
+   * Create a backup of the Cloud HSM Cluster in the specified subscription
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  backup is ArmResourceActionAsync<
+    CloudHsmCluster,
+    BackupRequestProperties,
+    ArmResponse<BackupResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = BackupResult> &
+      Azure.Core.Foundations.RetryAfterHeader,
+    OptionalRequestBody = true
+  >;
+
+  /**
+   * Queued validating pre restore operation
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  validateRestoreProperties is ArmResourceActionAsync<
+    CloudHsmCluster,
+    RestoreRequestProperties,
+    ArmResponse<RestoreResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = RestoreResult> &
+      Azure.Core.Foundations.RetryAfterHeader,
+    OptionalRequestBody = true
+  >;
+
+  /**
+   * Restores all key materials of a specified Cloud HSM Cluster
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header"
+  restore is ArmResourceActionAsync<
+    CloudHsmCluster,
+    RestoreRequestProperties,
+    ArmResponse<RestoreResult>,
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = RestoreResult> &
+      Azure.Core.Foundations.RetryAfterHeader
+  >;
+
+  /**
+   * Gets the private link resources supported for the Cloud Hsm Cluster.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @get
+  @action("privateLinkResources")
+  @operationId("CloudHsmClusterPrivateLinkResources_ListByCloudHsmCluster")
+  listByCloudHsmCluster is ArmResourceActionSync<
```

Comment:
Why isnt this operation in a CloudHsmClusterPrivateLinkResources interface?

Reply:
Because this operation operates on the resource CloudHsmCluster.